### PR TITLE
New from bicep example: 101/cyclecloud

### PR DESCRIPTION
**Pull request by StephenWeatherford**
Created a new QuickStart sample from the bicep example in https://github.com/Azure/bicep/tree/main/docs/examples/101/cyclecloud

Contribution guidelines: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/README.md

### CODE REVIEW

File: quickstarts/microsoft.compute/vm-cyclecloud/sub.bicep [hunk 1]
```diff
@@ -0,0 +1,6 @@
+targetScope = 'subscription'
+
+resource rg 'Microsoft.Resources/resourceGroups@2019-05-01' = {
+  name: 'rg-bicep'
+  location: 'eastus'
```
Comment (StephenWeatherford) [inline quickstarts/microsoft.compute/vm-cyclecloud/sub.bicep:5]: needs to be param

File: quickstarts/microsoft.compute/vm-cyclecloud/azuredeploy.parameters.json [hunk 2]
```diff
@@ -0,0 +1,9 @@
+{
+  "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentParameters.json#",
+  "contentVersion": "1.0.0.0",
+  "parameters": {
+    "ccadminRawPassword": {
+      "value": "TODO"
```
Comment (StephenWeatherford) [inline quickstarts/microsoft.compute/vm-cyclecloud/azuredeploy.parameters.json:6]: See https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/README.md#parameters-file-placeholders

File: quickstarts/microsoft.compute/vm-cyclecloud/main.bicep [hunk 3]
```diff
@@ -0,0 +1,189 @@
+param location string = resourceGroup().location
+param csadminSshKey string = 'ssh-rsa APublicSshKey2EAAAADAQABAAACAQCpXKIzGlb9fnLJmwrVPFQyntzOngz8QE4SOMHyXueVb4bO2IgSdHk8cD2LDToX3lDOUQzzWz+AiUO3iIRNJPOzZj/6aN+BJZXpi8+cdGOyaQmDyzk0T2w0mwTxXBk3DkXAw+lw13Q9SlFY+YLKsqHyKF7aXiy7RiJl4O3QUMGLKuGtmsqRFcrarp20pyH1UXQbvXUUoPU92AU4cSmx4AS8coWaDoQxWr7EA6toF0hgXsKFf/8MlkzJty0P7IhZ8KPzJg9lFNfBCUZHFEQWSb7FQBV6mFXxVcus1eCtoLEXkIDSkkYGd+edMO6t/Hc73c66M1vL9Ae6RUx2m39kZGF9bpVmcs8pZ2Hy2QukcGR8r61Jx913a32hRmk5fWpCnEo0NfE9XQJ7ibMNU97XL/QSeNZp3yzAyZqIYBkaYp8bFNjjMnVNyVdaANw2rjmxTY2XJlc0jVgucMWim8zT4YDQgKR8UuzXZBtC5uxlqhgZ8Zj+tRqgq/ZGo1MBacj89gQJjiiyFgf9hewVtdxlAEnDUHo0KI/Ro2geI+f2ylf1m0bUuPpwjO8sybJg/ZkA48LyOMbwj9wLHbiNJnPmoJROGGO/pAdvrzMlDuD/2BLf5Xmn3RkvnseS6/qCXZAfsrwlhb/LtCkWF2j+7e5EWXArhT14fGdWHi+5f09UY7ybAQ== user@Somewhere'
+param ccadminRawPassword string
+param myIp string = '20.49.199.4'
```
Comment (StephenWeatherford) [inline quickstarts/microsoft.compute/vm-cyclecloud/main.bicep:4]: no def for myIp?

File: quickstarts/microsoft.compute/vm-cyclecloud/azuredeploy.json [hunk 4]
```diff
@@ -0,0 +1,239 @@
...
+    }
+  },
+  "parameters": {
+    "location": {
+      "type": "string",
+      "defaultValue": "[resourceGroup().location]"
+    },
+    "csadminSshKey": {
+      "type": "string",
+      "defaultValue": "ssh-rsa APublicSshKey2EAAAADAQABAAACAQCpXKIzGlb9fnLJmwrVPFQyntzOngz8QE4SOMHyXueVb4bO2IgSdHk8cD2LDToX3lDOUQzzWz+AiUO3iIRNJPOzZj/6aN+BJZXpi8+cdGOyaQmDyzk0T2w0mwTxXBk3DkXAw+lw13Q9SlFY+YLKsqHyKF7aXiy7RiJl4O3QUMGLKuGtmsqRFcrarp20pyH1UXQbvXUUoPU92AU4cSmx4AS8coWaDoQxWr7EA6toF0hgXsKFf/8MlkzJty0P7IhZ8KPzJg9lFNfBCUZHFEQWSb7FQBV6mFXxVcus1eCtoLEXkIDSkkYGd+edMO6t/Hc73c66M1vL9Ae6RUx2m39kZGF9bpVmcs8pZ2Hy2QukcGR8r61Jx913a32hRmk5fWpCnEo0NfE9XQJ7ibMNU97XL/QSeNZp3yzAyZqIYBkaYp8bFNjjMnVNyVdaANw2rjmxTY2XJlc0jVgucMWim8zT4YDQgKR8UuzXZBtC5uxlqhgZ8Zj+tRqgq/ZGo1MBacj89gQJjiiyFgf9hewVtdxlAEnDUHo0KI/Ro2geI+f2ylf1m0bUuPpwjO8sybJg/ZkA48LyOMbwj9wLHbiNJnPmoJROGGO/pAdvrzMlDuD/2BLf5Xmn3RkvnseS6/qCXZAfsrwlhb/LtCkWF2j+7e5EWXArhT14fGdWHi+5f09UY7ybAQ== user@Somewhere"
```
Comment (StephenWeatherford) [inline quickstarts/microsoft.compute/vm-cyclecloud/azuredeploy.json:18]: ??

File: quickstarts/microsoft.compute/vm-cyclecloud/main.bicep (hunk 3 above)
Comment (MarcusFelling) [reply quickstarts/microsoft.compute/vm-cyclecloud/main.bicep:4]: Added description for this and moved value to param file.

File: quickstarts/microsoft.compute/vm-cyclecloud/azuredeploy.parameters.json (hunk 2 above)
Comment (MarcusFelling) [reply quickstarts/microsoft.compute/vm-cyclecloud/azuredeploy.parameters.json:6]: Added secure decorator and used GEN-PASSWORD for value.

File: quickstarts/microsoft.compute/vm-cyclecloud/azuredeploy.json (hunk 4 above)
Comment (MarcusFelling) [reply quickstarts/microsoft.compute/vm-cyclecloud/azuredeploy.json:18]: Removed default and used GEN-SSH-PUB-KEY.

**Review by MarcusFelling — COMMENTED** [review]
PR with updates here: https://github.com/Azure/azure-quickstart-templates/pull/11678

**Pull request closed without merging**